mitodl/mit-open
# Homepage hero section

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/3939

### Description (What does it do?)
This PR:
1. Adds the "hero search" section in homepage.
2. Makes several customizations to the `@mui/material` theme. These may change, but I wanted to get the setup in place.
    - Typography, to align with [typography designs](https://www.figma.com/file/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=104%3A822&mode=dev)
    - Some customizations to chips and inputs, but largely that stuff will be polished in a separate PR.
3. Updated storybook *That's most of the line changes*


### Screenshots (if appropriate):

<img width="1593" alt="Screenshot 2024-04-11 at 5 03 09 PM" src="https://github.com/mitodl/mit-open/assets/9010790/19813b77-74d9-422a-b002-c51633cf60a3">

<img width="491" alt="Screenshot 2024-04-11 at 4 54 48 PM" src="https://github.com/mitodl/mit-open/assets/9010790/968406b9-5679-4ae1-b2d4-fa900b618491">

https://github.com/mitodl/mit-open/assets/9010790/24d2f0f4-a7f3-4260-823d-13bda5ee1fb0


### How can this be tested?
1. Enter search text; hit enter or click search icon => you should search.
2. Click popular search links. They should take you to relevant searches.
3. Resize screen. Should look good.

**For review:** I suggest starting with usage (Homepage + HeroSearch components), then reviewing the MUI theme customizations in `frontends/ol-components/src/components/ThemeProvider/` directory.
